src/Symfony/Component/Form/Extension/Core/DataTransformer/DateTimeToArrayTransformer.php

```diff
@@ -72,14 +72,14 @@ public function transform($dateTime)
             ), array_flip($this->fields));
         }
 
-        if (!$dateTime instanceof \DateTime) {
-            throw new TransformationFailedException('Expected a \DateTime.');
+        if (!$dateTime instanceof \DateTime && !$dateTime instanceof \DateTimeInterface) {
+            throw new TransformationFailedException('Expected a \DateTime or \DateTimeInterface.');
         }
 
         $dateTime = clone $dateTime;
```

Comment:
Fixed.

I chose the more defensive `if (!$value instanceof \DateTimeImmutable) { clone …` over `if ($value instanceof \DateTime) { clone …` to be compatible with possible future non-immutable implementations of the interface (apparently you cannot implement this interface in your own PHP code, but future PHP versions may remove this restriction or add new native implementations).

I realized that in two of the five affected files, the setTimezone() call was useless (setting the time zone does not affect the output getTimestamp()), so I simply removed it.

Thanks for the thorough review :)